src/Service/SubsiteService.php

```diff
@@ -18,25 +19,28 @@ class SubsiteService {
   // Disable phpcs for a bit, so we don't have to add a load of stuff that's
   // made redundant by type hints.
   // phpcs:disable
+  private ConfigFactory $configFactory;
   private EntityTypeManagerInterface $entityTypeManager;
   private MenuLinkManagerInterface $menuLinkService;
+  private ModuleHandlerInterface $moduleHandler;
   private RouteMatchInterface $routeMatch;
-  private ConfigFactory $configFactory;
   private ?NodeInterface $subsiteHomePage;
   private bool $searched = false;
   private ?array $subsiteTypes = [];
   private ?string $themeField;
 
   public function __construct(
+    ConfigFactory $configFactory,
     EntityTypeManagerInterface $entityTypeManager,
     MenuLinkManagerInterface $menuLinkService,
+    ModuleHandlerInterface $moduleHandler,
     RouteMatchInterface $routeMatch,
-    ConfigFactory $configFactory
   ) {
+    $this->configFactory = $configFactory;
```

Comment:
Property promotion is much neater - you don't need to declare the property at the top or set it in the constructor e.g
```
  public function __construct(
    private ConfigFactory $configFactory
    private EntityTypeManagerInterface $entityTypeManager,
    private MenuLinkManagerInterface $menuLinkService,
    private ModuleHandlerInterface $moduleHandler,
    private RouteMatchInterface $routeMatch
) {}

```

Reply:
Good shout. I've copied over that bit from your MR :)